chitragupta.py

```diff
@@ -6,53 +6,55 @@
 import requests
 import json
 import pywikibot
+import grequests
 
 ignored_repos = set({'git-exercise'})
 
 def get_all_repos():
     # get a list of all repositories in the MetaKGP org
     r = requests.get("https://api.github.com/orgs/metakgp/repos")
 
+    all_contribs = set()
+
     if not r.ok:
         print "Looks like something is wrong"
         print r.status_code
         exit(0)
     else:
         repos = json.loads(r.text)
-        contributor_urls = {}
-        for repo in repos:
-            if repo['name'] not in ignored_repos:
-                contributor_urls[repo['name']] = repo['contributors_url']
-        contributors = {}
-        for repo, url in contributor_urls.items():
-            try:
-                print "Getting ", repo
-                # ignore the sandbox
-                contributor_response = requests.get(url)
-                contributors[repo] = set()
-                repo_contributors = json.loads(contributor_response.text)
-                for contributor in repo_contributors:
-                    contributors[repo].add(contributor['login'])
-            except Exception as e:
-                print "Failed due to ", str(e)
-                pass
-
-    consolidated_list = set()
-    for repo in contributors:
-        print repo, contributors[repo]
-        consolidated_list = consolidated_list.union(contributors[repo])
-    return contributors, consolidated_list
+
+        contrib_urls = [repo['contributors_url'] for repo in repos if repo['name'] not in ignored_repos]
+
+        all_reqs = (grequests.get(u) for u in contrib_urls)
+
+        all_resps = grequests.map(all_reqs)
```

Comment:
Shouldn't we handle exceptions here? Or is already taken care of by the library?

Reply:
Yeah @zorroblue we should handle exceptions here. How would you like to handle exceptions?

Reply:
I feel we could sleep for a second and do one more retry before going onto the next one. What say, @icyflame ?

Reply:
You are talking about request retries, I think there should be a configuration option for that in requests. I will check it out. Better to let the library handle that instead of manually trying to. 5 retries is standard, and generally if a page isn't available five times, its probably unavailable. What do you think?

Reply:
Sure @icyflame ! If you can include that, it would be great! :smile:  If we included request retries, we could *safely* assume our data would be correct.